### PR TITLE
Adds unit tests in CI on windows runner

### LINKED ISSUE #160: Add support for Windows test in e2e test suite
### Description of problem

Currently CI does not support Installing and testing Apps plugin functionality on Windows OS. This story is about adding support for that feature.

### Actions:
- Add a Windows OS runner in Github Actions CI
- Add Installation tests in Windows OS
- Run Integration test suite on Windows OS

### PR DESCRIPTION
**Pull request by shaheerkootteeri**
Pull request

### What this PR does / why we need it

PR adds unit tests on windows runner in CI.
### Which issue(s) this PR fixes

Fixes #160 

### Describe testing done for PR

CI commands have been ran successfully in windows as well as ubuntu runner.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 
 jobs:
 
-  unit:
+  unitubuntu:
```
Comment (odinnordico) [inline .github/workflows/ci.yaml:14]:
```suggestion
  unit-ubuntu:
```
use dash (`-`) to separate words

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -68,8 +68,70 @@ jobs:
         git diff --exit-code docs
       continue-on-error: true
 
+  unitwindows:
```
Comment (odinnordico) [inline .github/workflows/ci.yaml:71]:
```suggestion
  unit-windows:
```
use dash (-) to separate words

File: Makefile [hunk 3]
```diff
@@ -46,7 +46,7 @@ install: publish-local ## Install the plugin binaries to the local machine
 
 .PHONY: build-local
 build-local: prepare test ## Build the plugin binaries locally for the host architecture
-	@echo BUILD_VERSION: $(BUILD_VERSION)
+    echo BUILD_VERSION: $(BUILD_VERSION);
```
Comment (odinnordico) [inline Makefile:49]: does the `echo` works in linux/windows/mac without the `@`?

Comment (shaheerkootteeri) [reply Makefile:49]: Below screens from linux and windows runner as well as my mac machine where I ran make commands to see if it picks up the right version and it seems to me working.

linux 
image

Wndows
image

mac os 
image

Comment (odinnordico) [reply Makefile:49]: As the `@` is intended to not _echo_ the output of the commend, i see value removing the `@`

**Review by warango4 — APPROVED** [review]

**Pull request merged**